src/VisualStudio/Core/Def/ErrorReporting/VisualStudioInfoBar.cs

```diff
@@ -4,121 +4,164 @@
+    IVsWindowFrame? windowFrame)
 {
-    private readonly IThreadingContext _threadingContext;
-    private readonly SVsServiceProvider _serviceProvider;
-    private readonly IAsynchronousOperationListener _listener;
+    private readonly IThreadingContext _threadingContext = threadingContext;
+    private readonly SVsServiceProvider _serviceProvider = serviceProvider;
+    private readonly IAsynchronousOperationListener _listener = listenerProvider.GetListener(FeatureAttribute.InfoBar);
+    private readonly IVsWindowFrame? _windowFrame = windowFrame;
 
     /// <summary>
-    /// Keep track of the messages that are currently being shown to the user.  If we would 
-    /// show the same message again, block that from happening so we don't spam the user with
-    /// the same message.  When the info bar item is dismissed though, we then may show the
-    /// same message in the future.  This is important for user clarity as it's possible for 
-    /// a feature to fail for some reason, then work fine for a while, then fail again.  We want
-    /// the second failure message to be reported to ensure the user is not confused.
+    /// Keep track of the messages that are currently being shown to the user.  If we would show the same message again,
+    /// block that from happening so we don't spam the user with the same message.  When the info bar item is dismissed
+    /// though, we then may show the same message in the future.  This is important for user clarity as it's possible
+    /// for a feature to fail for some reason, then work fine for a while, then fail again.  We want the second failure
+    /// message to be reported to ensure the user is not confused.
     /// 
-    /// Accessed on UI thread.
+    /// Accessed on UI thread only.
     /// </summary>
     private readonly HashSet<string> _currentlyShowingMessages = [];
 
-    public VisualStudioInfoBar(
-        IThreadingContext threadingContext,
-        SVsServiceProvider serviceProvider,
-        IAsynchronousOperationListenerProvider listenerProvider)
-    {
-        _threadingContext = threadingContext;
-        _serviceProvider = serviceProvider;
-        _listener = listenerProvider.GetListener(FeatureAttribute.InfoBar);
-    }
+    public InfoBarMessage? ShowInfoBarMessage(string message, params InfoBarUI[] items)
+        => ShowInfoBarMessage(message, isCloseButtonVisible: true, KnownMonikers.StatusInformation, items);
 
-    public void ShowInfoBar(string message, params InfoBarUI[] items)
+    public InfoBarMessage? ShowInfoBarMessage(
+        string message,
+        bool isCloseButtonVisible,
+        ImageMoniker imageMoniker,
+        params InfoBarUI[] items)
     {
         // We can be called from any thread since errors can occur anywhere, however we can only construct and InfoBar from the UI thread.
-        _threadingContext.JoinableTaskFactory.RunAsync(async () =>
+        return _threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
@sharwell  to make sure this is ok.

Reply:
Code was restored to prior form, so did not review.